restdocs-api-spec-postman-generator/src/main/kotlin/com/epages/restdocs/apispec/postman/PostmanCollectionGenerator.kt

```diff
@@ -73,41 +73,53 @@ object PostmanCollectionGenerator {
                 }
             }
             header = modelsWithSamePathAndMethod
-                    .flatMap { it.request.headers }
-                    .distinctBy { it.name }
-                    .toItemHeader(modelsWithSamePathAndMethod.map { it.request.contentType }.firstOrNull())
-                    .ifEmpty { null }
+                .flatMap { it.request.headers }
+                .distinctBy { it.name }
+                .toItemHeader(modelsWithSamePathAndMethod.map { it.request.contentType }.firstOrNull())
+                .ifEmpty { null }
         }
     }
 
     private fun toUrl(modelsWithSamePathAndMethod: List<ResourceModel>, url: String): Url {
-        val baseUri = URI(url)
+        val urlStartWithVariable = url.startsWith("{{")
+        val baseUrl = when (urlStartWithVariable) {
+            true -> URL("http://$url")
+            else -> URL(url)
+        }
+
         return Url().apply {
-            protocol = baseUri.scheme
-            host = baseUri.host
-            port = when (baseUri.port) {
-                    -1 -> null
-                    else -> baseUri.port.toString()
+            protocol = when (urlStartWithVariable) {
+                true -> null
+                else -> baseUrl.protocol
+            }
+            host = baseUrl.host
+            port = when (baseUrl.port) {
+                -1 -> null
+                else -> baseUrl.port.toString()
             }
-            path = baseUri.path + modelsWithSamePathAndMethod.first().request.path.replace(Regex("\\{[^/]+}")) {
+            path = baseUrl.path + modelsWithSamePathAndMethod.first().request.path.replace(Regex("(?<!\\{)\\{([^}]+)\\}(?!\\})")) {
```

Comment:
Can you explain why the change in the regex is needed? Can you also please add a comment explaining what it does.

Reply:
With the current regex, if we allow the postman variables `{{foo}}` in the path, it will replaced to `{:foo}`. So the new regex identify the difference between one and two curly brackets and only replace one curly bracket to colon.
E.g. `{foo}` will be replaced to `:foo` and `{{bar}}` stays the same.

See the test `should allow postman variable as part of the path` in https://github.com/ePages-de/restdocs-api-spec/pull/99/files/45969c105da0219eabff6cda2ae53907353a3a8b#diff-5ddef866bfe4db14f10c446ba410269dR123

Reply:
Sounds very reasonable. Thanks.